landscapist-zoomable/src/appleMain/kotlin/com/skydoves/landscapist/zoomable/subsampling/ImageRegionDecoder.kt

```
@@ -0,0 +1,409 @@
+
+          try {
+            @Suppress("UNCHECKED_CAST")
+            val propsDict = CFBridgingRelease(properties) as? Map<Any?, Any?>
+            val width = (propsDict?.get(kCGImagePropertyPixelWidth) as? Number)?.toInt()
+            val height = (propsDict?.get(kCGImagePropertyPixelHeight) as? Number)?.toInt()
+
+            if (width != null && height != null && width > 0 && height > 0) {
+              return IntSize(width, height)
+            }
+            return null
+          } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)